pkg/services/caching/util.go

```diff
@@ -0,0 +1,74 @@
+package caching
```

Comment:
With the middleware caching impl, we had direct access to the HTTP response and could write the headers ourselves. I had to come up with a workaround, so I chose this. 

It is not ideal, but modifying the `QueryData` function would make the `query.go` file messy since there are some recursive calls to that function. 

Modifying `QueryDataResponse` also didn't seem like a good option, as a _lot_ of code uses that object. 

Looking for a "good enough", or a recommendation for a better approach